### ISSUE TITLE
Warning due to schema reference in ibm-web-ext.xml

### ISSUE THREAD

**Issue by kwsutter** [WASdev/tool.accelerate.core#67]
I am getting the following warning due to the schema reference in the ibm-web-ext.xml file…

schema_reference.4: Failed to read schema document 'http://websphere.ibm.com/xml/ns/javaee/ibm-web-ext_1_0.xsd', because 
1) could not find the document; 2) the document could not be read; 3) the root element of the document is not <xsd:schema>.

This warning is appearing after importing my App Accelerator project into Eclipse.

**Comment by MarkNSweep** [comment]
AFAIK the requirement for a schema URL is that is unique, so that it provides a namespace, rather than it actually being an accessible web endpoint. For convenience (and quite possibly best practice
), lots of people do host schema docs at these URLs, enough so that tools like Eclipse expect them to be there. We can't change the schema location as it's already in use, so we'd have to host it at this address to make the warning go away.

**Comment by kwsutter** [comment]
Am I the only one getting this warning message?  Or, am I doing something unique to get this warning while most users are not?  It just seems that for the new users we're targeting the App Accelerator for we should have an initial "clean" experience.

**Comment by katheris** [comment]
@kwsutter I believe anyone who references this schema location will see this problem. My understanding is that this is the standard location to reference when creating an ibm-web-ext.xm file. I agree that we should strive for a clean experience. We have two options: delete the location from the xml file (which prevents validation but doesn't break the deployment) or actually host the schema somewhere. If we go the hosting route we should probably coordinate with WDT etc and make sure we are all referencing the same place. Do you have a preference?

**Comment by ebullient** [comment]
why do we need the ibm-web-ext file? surely we can just leave it out.

**Comment by iainduncani** [comment]
@ebullient we set the context root in it for some reason though I think this would be better in the server.xml.

**Comment by MarkNSweep** [comment]
I believe it was to cope with the fact that the build will generated a war file which contains the version number in the name i.e. app.x.y.z.war. Setting it in ivm-web-ext allowed the build to just generate a war that was always mapped to the correct context root. If you specify the setting in server.xml you either have to know the name of the file that will be generated or change it when it's deployed into the server. I'm not saying I think we should keep it, but this is why I believe that we used this file.

**Comment by NottyCode** [comment]
@kwsutter do you have WDT installed? I ask because I do and with WDT installed I don't get this warning. WDT updates the eclipse XML Catalog with the right schema. If you don't you can still get rid of these warnings in Eclipse by adding to them to the XML Catalog through preferences. This can be done by going Window > Preferences > XML > XML Catalog and clicking Add. Eclipse only tries the schema location url if it isn't in its catalog. In that case the only fix as @MarkNSweep says would be for the url to be resolvable.

**Comment by kwsutter** [comment]
No, I didn't have WDT installed for this sample.  I don't think we should have that as a requirement for app accelerator.  And, even the proposed workaround sounds messy.  We want a pleasant experience for our newbie app accelerator users.

**Comment by NottyCode** [comment]
This isn't an app accelerator issue, this is an issue with the schema. Even if app accelerator stopped producing this file, as soon as someone needed to use it and created it the warning would come back unless they have WDT. I agreen with @MarkNSweep there isn't a change that can be made to the app accelerator for this.
